modules/core/04-channel/keeper/grpc_query.go

```diff
@@ -335,8 +335,18 @@ func (q Keeper) PacketAcknowledgements(c context.Context, req *types.QueryPacket
 			return err
 		}
 
-		ack := types.NewPacketState(req.PortId, req.ChannelId, sequence, value)
-		acks = append(acks, &ack)
+		if len(req.Commitments) > 0 {
+			for _, commitment := range req.Commitments {
+				if sequence == commitment.Sequence {
+					ack := types.NewPacketState(req.PortId, req.ChannelId, sequence, value)
+					acks = append(acks, &ack)
+				}
+			}
+		} else {
```

Comment:
Assuming we want to preserve previous functionality if no commitments are included in the req

Reply:
Maybe add a comment for future readers?

Reply:
Done!